hack/make-rules/test.sh

```diff
@@ -71,7 +71,10 @@ KUBE_COVER_REPORT_DIR="${KUBE_COVER_REPORT_DIR:-}"
 # How many 'go test' instances to run simultaneously when running tests in
 # coverage mode.
 KUBE_COVERPROCS=${KUBE_COVERPROCS:-4}
-KUBE_RACE=${KUBE_RACE:-}   # use KUBE_RACE="-race" to enable race testing
+# use KUBE_RACE="" to disable the race detector
+# this is defaulted to "-race" in make test as well
+# NOTE: DO NOT ADD A COLON HERE. KUBE_RACE="" is meaningful!
```

Comment:
we usually use `${FOO:-default}` but this will default in the set-but-null case. We need `${FOO-default}` here (which will not override set-but-empty) so you can get the previous behavior with `KUBE_RACE=""` from the commandline/env/...

Reply:
make this the code comment?
or add a link to https://wiki.bash-hackers.org/syntax/pe#use_a_default_value?

Reply:
I tried to not make this overly verbose, hopefully future editor will think twice before changing, we have history to point back here / reviewers should check that the result is sane.

it will also break CI if done wrong (which is why this PR took so long)